### PR TITLE
Source maps support

### LINKED ISSUE #53: Source Maps not working with Webpack 4.8
My version of optimize-css-assets-webpack-plugin: 4.0.1

The issue I'm facing is the same as described here: 
https://github.com/webpack-contrib/mini-css-extract-plugin/issues/141

In summary, I'm defining source-maps on my configuration. I'm also using MiniCssExtractPlugin to extract the css modules to a specific bundle, but when using `optimize-css-assets-webpack-plugin` on webpacks 4 `optimization.minimizer`, the source-maps are not generated. Here is a snippet:

```
  optimization: {
    minimizer: [
      new UglifyJsPlugin({
        cache: true,
        parallel: true,
        sourceMap: true // set to true if you want JS source maps
      }),
      new OptimizeCSSAssetsPlugin({}) // <---- THIS. Commenting/Uncommenting results in: No Optimization with source-maps/ Optimization without source-maps
    ]
  },

  plugins: [

    new webpack.DefinePlugin({
      'process.env.NODE_ENV': 'production'
    }),
    new MiniCssExtractPlugin({
      filename: 'foo.min.css'
    }),
  ],
  devtools: 'source-maps'
```

Maybe optimize-css-assets-webpack-plugin is truncating source-maps some how?!

### PR DESCRIPTION
**Pull request by frenzzy**
Fixes #53

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -30,33 +30,36 @@ class OptimizeCssAssetsWebpackPlugin extends LastCallWebpackPlugin {
   }
 
   processCss(assetName, asset, assets) {
-    const css = asset.source();
+    const css = asset.sourceAndMap ? asset.sourceAndMap() : { source: asset.source() };
     const processOptions = Object.assign(
       { from: assetName, to: assetName },
-      this.options.cssProcessorOptions || {}
+      this.options.cssProcessorOptions
```
Comment (frenzzy) [inline src/index.js]: From MDN:

> Note that [Object.assign()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) does not throw on `null` or `undefined` source values.

File: src/index.js [hunk 2]
```diff
@@ -30,33 +30,36 @@ class OptimizeCssAssetsWebpackPlugin extends LastCallWebpackPlugin {
   }
 
   processCss(assetName, asset, assets) {
-    const css = asset.source();
+    const css = asset.sourceAndMap ? asset.sourceAndMap() : { source: asset.source() };
```
Comment (frenzzy) [inline src/index.js]: [UglifyJS Webpack Plugin](https://github.com/webpack-contrib/uglifyjs-webpack-plugin/blob/v1.2.6/src/index.js#L137) also uses `sourceAndMap` function if exists

File: src/index.js [hunk 3]
```diff
@@ -30,33 +30,36 @@ class OptimizeCssAssetsWebpackPlugin extends LastCallWebpackPlugin {
...
           }
         }
+        if (
+          map &&
+          (
+            (map.sources && map.sources.length > 0) ||
+            (map.mappings && map.mappings.length > 0)
+          )
+        ) {
+          processOptions.map = Object.assign({ prev: map }, processOptions.map);
```
Comment (frenzzy) [inline src/index.js]: From [PostCSS docs](https://github.com/postcss/postcss/blob/6.0.23/docs/source-maps.md):

> `prev` string, object, boolean or function: source map content from a previous processing step (for example, Sass compilation). PostCSS will try to read the previous source map automatically (based on comments within the source CSS), but you can use this option to identify it manually. If desired, you can omit the previous map with `prev: false`.

**Comment by NMFR** [comment]
Thx for the help

**Pull request merged**